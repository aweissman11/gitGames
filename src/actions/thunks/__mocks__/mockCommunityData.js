export const mockCommunityData = {
  "followers": {
    "totalCount": 14
  },
  "following": {
    "totalCount": 5
  },
  "starredRepositories": {
    "totalCount": 16
  },
  "repositories": {
    "totalCount": 64
  },
  "repositoriesContributedTo": {
    "totalCount": 9
  },
  "watching": {
    "totalCount": 1
  },
  "issues": {
    "totalCount": 92
  },
  "issueComments": {
    "totalCount": 35
  },
  "pullRequests": {
    "totalCount": 233
  }
}