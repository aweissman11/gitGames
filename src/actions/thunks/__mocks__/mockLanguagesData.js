export const mockLanguagesData = {
  "Overall": {
    "Ruby": 0.6776647042377972,
    "JavaScript": 0.05299369091826422,
    "CSS": 0.10368481305012495,
    "HTML": 0.08045850211897976,
    "CoffeeScript": 0.0002282019793006745,
    "Elixir": 0.08344675836925351,
    "Python": 0.0014611415831052665,
    "Shell": 0.00006218774317435442
  },
  "Repositories": [
    {
      "Ruby": 0.6755172413793104,
      "JavaScript": 0.04103448275862069,
      "CSS": 0.008620689655172414,
      "HTML": 0.27482758620689657,
      "name": "evening-ajax-brussels-sprouts"
    },
    {
      "CSS": 0.1651287032540068,
      "JavaScript": 0.20106847984458476,
      "Ruby": 0.2967459932005828,
      "HTML": 0.3370568237008256,
      "name": "infinite-brussel-sprouts"
    },
    {
      "Ruby": 1,
      "name": "csv-to-db"
    },
    {
      "Ruby": 0.4706527701590784,
      "HTML": 0.5293472298409215,
      "name": "news-aggregator-postgresql"
    },
    {
      "Ruby": 1,
      "name": "korning"
    },
    {
      "Ruby": 0.5212305720237141,
      "HTML": 0.47876942797628586,
      "name": "movie-catalog-deluxe"
    },
    {
      "Ruby": 1,
      "name": "blackjack"
    },
    {
      "Ruby": 1,
      "name": "oo-marathon"
    },
    {
      "Ruby": 0.597636921224436,
      "HTML": 0.39335101879550716,
      "CSS": 0.00901205998005682,
      "name": "meetups-in-space"
    },
    {
      "Ruby": 0.8637429459347268,
      "JavaScript": 0.007157842774281131,
      "CoffeeScript": 0.004734497885182817,
      "CSS": 0.023291036989667127,
      "HTML": 0.10107367641614216,
      "name": "rescue_mission"
    },
    {
      "Ruby": 0.4160206489152224,
      "JavaScript": 0.00626208511468663,
      "CSS": 0.4668623937840379,
      "HTML": 0.11085487218605311,
      "name": "pull-request-practice"
    },
    {
      "Ruby": 0.8972313613507977,
      "JavaScript": 0.010304798559811286,
      "CSS": 0.008473524116953256,
      "HTML": 0.08399031597243777,
      "name": "catstagram"
    },
    {
      "Ruby": 0.4887849953919301,
      "JavaScript": 0.008062596510073667,
      "CSS": 0.33810416069041704,
      "HTML": 0.1650482474075792,
      "name": "Season-Survey"
    },
    {
      "Ruby": 0.5359934200296262,
      "JavaScript": 0.05881871526896856,
      "CSS": 0.2555794711471573,
      "HTML": 0.14960839355424793,
      "name": "Brewers_Board"
    },
    {
      "Ruby": 0.2689907659764878,
      "HTML": 0.13972257250945774,
      "JavaScript": 0.46441036488630355,
      "CSS": 0.12687629662775088,
      "name": "em-websocket-chat-demo"
    },
    {
      "Ruby": 0.6315856290465913,
      "JavaScript": 0.15455485228241353,
      "CSS": 0.03911931663613776,
      "HTML": 0.17474020203485735,
      "name": "Hiking-Buddy"
    },
    {
      "Elixir": 1,
      "name": "blackjack-elixir"
    },
    {
      "Python": 1,
      "name": "BBC-Article-Summarizer"
    },
    {
      "Elixir": 1,
      "name": "address_us"
    },
    {
      "Elixir": 1,
      "name": "simple_bayes"
    },
    {
      "JavaScript": 0.7615300328718,
      "HTML": 0.23538201016037455,
      "CSS": 0.0030879569678254806,
      "name": "ember-simple-auth-example"
    },
    {
      "JavaScript": 0.7856194957172156,
      "HTML": 0.2143805042827844,
      "name": "ember-recurly-service"
    },
    {
      "Ruby": 1,
      "name": "jsonapi-resources"
    },
    {
      "Ruby": 0.9989315345950516,
      "Shell": 0.0010684654049483884,
      "name": "jsonapi-utils"
    }
  ]
}