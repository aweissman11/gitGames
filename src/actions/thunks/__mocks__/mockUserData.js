export const mockUserData = {
  "avatarUrl": "https://previews.123rf.com/images/alexraths/alexraths1501/alexraths150100053/35560687-portrait-of-a-screaming-clown-isolated-on-white-background.jpg",
  "createdAt": "2015-05-26T20:30:19Z",
  "email": "",
  "login": "npendery",
  "name": "Nick Pendery"
}