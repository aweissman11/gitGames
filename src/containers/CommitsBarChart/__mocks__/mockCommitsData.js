export const mockCommitsData = [
  {
    "firstDay": "2017-12-31",
    "contributionDays": [
      {
        "date": "2017-12-31",
        "contributionCount": 0
      },
      {
        "date": "2018-01-01",
        "contributionCount": 0
      },
      {
        "date": "2018-01-02",
        "contributionCount": 0
      },
      {
        "date": "2018-01-03",
        "contributionCount": 0
      },
      {
        "date": "2018-01-04",
        "contributionCount": 0
      },
      {
        "date": "2018-01-05",
        "contributionCount": 0
      },
      {
        "date": "2018-01-06",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  },
  {
    "firstDay": "2018-01-07",
    "contributionDays": [
      {
        "date": "2018-01-07",
        "contributionCount": 0
      },
      {
        "date": "2018-01-08",
        "contributionCount": 0
      },
      {
        "date": "2018-01-09",
        "contributionCount": 0
      },
      {
        "date": "2018-01-10",
        "contributionCount": 0
      },
      {
        "date": "2018-01-11",
        "contributionCount": 0
      },
      {
        "date": "2018-01-12",
        "contributionCount": 0
      },
      {
        "date": "2018-01-13",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  },
  {
    "firstDay": "2018-01-14",
    "contributionDays": [
      {
        "date": "2018-01-14",
        "contributionCount": 0
      },
      {
        "date": "2018-01-15",
        "contributionCount": 0
      },
      {
        "date": "2018-01-16",
        "contributionCount": 0
      },
      {
        "date": "2018-01-17",
        "contributionCount": 0
      },
      {
        "date": "2018-01-18",
        "contributionCount": 0
      },
      {
        "date": "2018-01-19",
        "contributionCount": 0
      },
      {
        "date": "2018-01-20",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  }
]