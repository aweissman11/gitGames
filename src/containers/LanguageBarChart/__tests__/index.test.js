import React from 'react';
import { shallow } from 'enzyme';

import { LanguageBarChart } from '../index';

import { mockLanguageData } from '../__mocks__/mockLanguageData';

describe('LanguageBarChart', () => {

  let wrapper;
  let mockLoadingLanguages;
  
  beforeEach(() => {
  })

  it('should match the snapshot', () => {
    mockLoadingLanguages = {
      loadingLanguages: false,
      message: 'not loading'
    }
    wrapper = shallow(<LanguageBarChart
                        languageData={mockLanguageData}
                        loadingLanguages={mockLoadingLanguages}
                      />)
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should match the snapshot', () => {
    mockLoadingLanguages = {
      loadingLanguages: true,
      message: 'is loading'
    }
    wrapper = shallow(<LanguageBarChart
                        languageData={{}}
                        loadingLanguages={mockLoadingLanguages}
                      />)
    expect(wrapper).toMatchSnapshot()
  })
  
  it('should match the snapshot', () => {
    mockLoadingLanguages = {
      loadingLanguages: true,
      message: 'is loading'
    }
    wrapper = shallow(<LanguageBarChart
                        languageData={{}}
                        loadingLanguages={mockLoadingLanguages}
                      />)
    expect(wrapper).toMatchSnapshot()
  })
 
  it('should clean the data', () => {
    wrapper = shallow(<LanguageBarChart
                        languageData={mockLanguageData}
                        loadingLanguages={mockLoadingLanguages}
                      />)
    const results = wrapper.instance().cleanedUpData()
    const expected = {
      "Ruby": "32.63",
      "hover": "percent used",
      "CSS": "2.71",
      "HTML": "8.55",
      "Roff": "0.14",
      "JavaScript": "55.95",
      "CoffeeScript": "0.01"
    }

    expect(results).toEqual(expected);
  }) 
})
