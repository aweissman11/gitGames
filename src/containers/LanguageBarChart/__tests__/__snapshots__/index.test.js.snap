// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageBarChart should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LanguageBarChart
    languageData={
      Object {
        "Overall": Object {
          "CSS": 0.027094665090688645,
          "CoffeeScript": 0.00013974515605317292,
          "HTML": 0.08554854057929996,
          "JavaScript": 0.5595484354944749,
          "Roff": 0.0013872963043256535,
          "Ruby": 0.3262813173751577,
        },
        "Repositories": Array [
          Object {
            "Ruby": 1,
            "name": "credit_check",
          },
          Object {
            "Ruby": 1,
            "name": "sorting_cards",
          },
          Object {
            "Ruby": 1,
            "name": "morse_translator",
          },
          Object {
            "Ruby": 1,
            "name": "enigma",
          },
          Object {
            "Ruby": 1,
            "name": "scrabble",
          },
          Object {
            "Ruby": 1,
            "name": "wizarding_bank",
          },
          Object {
            "CSS": 0.7837843933243122,
            "HTML": 0.12539467749210645,
            "Ruby": 0.09082092918358142,
            "name": "denver_public_library",
          },
          Object {
            "Ruby": 1,
            "name": "reunion",
          },
          Object {
            "Ruby": 1,
            "name": "pantry",
          },
          Object {
            "Ruby": 1,
            "name": "Award_Show",
          },
          Object {
            "Ruby": 1,
            "name": "SuperFizzBuzz",
          },
          Object {
            "Ruby": 1,
            "name": "municipal-budget",
          },
          Object {
            "Ruby": 1,
            "name": "credit_check2",
          },
          Object {
            "Ruby": 1,
            "name": "sorting_cards2",
          },
          Object {
            "Ruby": 1,
            "name": "dog_party",
          },
          Object {
            "Ruby": 1,
            "name": "binary_translator",
          },
          Object {
            "Ruby": 1,
            "name": "night_writer",
          },
          Object {
            "Ruby": 1,
            "name": "potluck",
          },
          Object {
            "Ruby": 1,
            "name": "connect_four",
          },
          Object {
            "Ruby": 1,
            "name": "world_cup",
          },
          Object {
            "Ruby": 1,
            "name": "black_thursday",
          },
          Object {
            "Ruby": 1,
            "name": "pantry-1",
          },
          Object {
            "Ruby": 1,
            "name": "market",
          },
          Object {
            "JavaScript": 0.5700892857142857,
            "Roff": 0.400765306122449,
            "Ruby": 0.029145408163265305,
            "name": "portfolios",
          },
          Object {
            "CSS": 0.010835913312693499,
            "HTML": 0.1715170278637771,
            "Ruby": 0.8176470588235294,
            "name": "personal_site",
          },
          Object {
            "CSS": 0.4103537877273636,
            "HTML": 0.5896462122726365,
            "name": "static_challenges",
          },
          Object {
            "HTML": 0.28995123912934195,
            "Ruby": 0.710048760870658,
            "name": "little-shop-redux",
          },
          Object {
            "CSS": 0.0077940545181369395,
            "CoffeeScript": 0.008491629104958145,
            "HTML": 0.10906310367031552,
            "JavaScript": 0.015507619660871432,
            "Ruby": 0.859143593045718,
            "name": "activerecord-obstacle-course",
          },
          Object {
            "HTML": 0.01481758187025277,
            "Ruby": 0.9851824181297473,
            "name": "laugh_tracks",
          },
          Object {
            "CSS": 0.4290060851926978,
            "HTML": 0.5709939148073022,
            "name": "set-list-song-index-B2",
          },
          Object {
            "CSS": 0.045615014859285555,
            "HTML": 0.13449149330833515,
            "JavaScript": 0.02259808125735485,
            "Ruby": 0.7972954105750244,
            "name": "jukebox",
          },
          Object {
            "CSS": 0.06989481546026016,
            "HTML": 0.16887616439492956,
            "JavaScript": 0.0064106553805937634,
            "Ruby": 0.7548183647642165,
            "name": "job-tracker",
          },
          Object {
            "CSS": 0.017162083656080556,
            "HTML": 0.14061289697908597,
            "JavaScript": 0.027425445391169635,
            "Ruby": 0.8147995739736639,
            "name": "apollo_14",
          },
          Object {
            "HTML": 0.14212652056496905,
            "JavaScript": 0.027077408407618957,
            "Ruby": 0.830796071027412,
            "name": "blogger",
          },
          Object {
            "CSS": 0.013841978875851702,
            "HTML": 0.14263680912125887,
            "JavaScript": 0.02211983366197458,
            "Ruby": 0.8214013783409149,
            "name": "school",
          },
          Object {
            "CSS": 0.08808202402355454,
            "HTML": 0.1550848120065661,
            "JavaScript": 0.014760676411579248,
            "Ruby": 0.7420724875583001,
            "name": "Plant_Pal",
          },
          Object {
            "CSS": 0.015811069979037508,
            "HTML": 0.13746041657374783,
            "JavaScript": 0.02526649123589492,
            "Ruby": 0.8214620222113197,
            "name": "vending-machine-tracker",
          },
          Object {
            "CSS": 0.001986047261875958,
            "HTML": 0.2273973707557061,
            "JavaScript": 0.006018630534720542,
            "Ruby": 0.7645979514476974,
            "name": "bike_share",
          },
          Object {
            "CSS": 0.1674861026491161,
            "HTML": 0.5953733747065852,
            "JavaScript": 0.0053511294992230705,
            "Ruby": 0.2317893931450756,
            "name": "jukebox_1806_final",
          },
          Object {
            "HTML": 0.0024107427478482625,
            "Ruby": 0.9975892572521518,
            "name": "rales_engine",
          },
          Object {
            "CSS": 0.01777970511708586,
            "HTML": 0.08922066658406641,
            "JavaScript": 0.021323256102093915,
            "Ruby": 0.8716763721967538,
            "name": "battleshift",
          },
          Object {
            "CSS": 0.0019241468842729971,
            "HTML": 0.152285793768546,
            "JavaScript": 0.014790430267062315,
            "Ruby": 0.8309996290801187,
            "name": "module_3_diagnostic",
          },
          Object {
            "CSS": 0.001986047261875958,
            "HTML": 0.2273973707557061,
            "JavaScript": 0.006018630534720542,
            "Ruby": 0.7645979514476974,
            "name": "bike_share",
          },
          Object {
            "CSS": 0.014890099956091639,
            "HTML": 0.2626882248108066,
            "JavaScript": 0.02866957667174626,
            "Ruby": 0.6937520985613554,
            "name": "Trelora",
          },
          Object {
            "CSS": 0.00186542005663685,
            "HTML": 0.1484020317346159,
            "JavaScript": 0.014339012001618195,
            "Ruby": 0.8353935362071291,
            "name": "module_3_diagnostic-1",
          },
          Object {
            "CSS": 0.02871174145168521,
            "HTML": 0.12558584994090558,
            "JavaScript": 0.023087582018991727,
            "Ruby": 0.8226148265884174,
            "name": "module_3_final_assessment",
          },
          Object {
            "CSS": 0.012950008219328207,
            "HTML": 0.10319823193117683,
            "JavaScript": 0.021589435423478055,
            "Ruby": 0.8622623244260169,
            "name": "Congress",
          },
          Object {
            "CSS": 0.36320677855629785,
            "HTML": 0.2774971484438651,
            "JavaScript": 0.35929607299983707,
            "name": "number_guesser",
          },
          Object {
            "JavaScript": 1,
            "name": "sorting_suite",
          },
          Object {
            "CSS": 0.24806583204388802,
            "HTML": 0.24405682937121959,
            "JavaScript": 0.5078773385848924,
            "name": "persephone",
          },
          Object {
            "CSS": 0.18530030604437644,
            "HTML": 0.20968821729150727,
            "JavaScript": 0.6050114766641163,
            "name": "RepresentativeTracker",
          },
          Object {
            "CSS": 0.0020656479909451045,
            "HTML": 0.0018018242950830587,
            "JavaScript": 0.9961325277139719,
            "name": "Play-FE",
          },
          Object {
            "JavaScript": 1,
            "name": "play-be",
          },
          Object {
            "CSS": 0.5882465911273286,
            "JavaScript": 0.4117534088726714,
            "name": "word_watch",
          },
          Object {
            "HTML": 0.0047756245806528,
            "Ruby": 0.9952243754193472,
            "name": "GitGames-BackEnd",
          },
          Object {
            "CSS": 0.1057339060645453,
            "HTML": 0.0266348273328435,
            "JavaScript": 0.8676312666026113,
            "name": "gitGames",
          },
          Object {
            "CSS": 0.01481496959692417,
            "HTML": 0.14344818939758028,
            "JavaScript": 0.024217982740246984,
            "Ruby": 0.8175188582652485,
            "name": "promises",
          },
          Object {
            "HTML": 0.007872734962100263,
            "Ruby": 0.9921272650378997,
            "name": "gitGames_be_messaging",
          },
        ],
      }
    }
    loadingLanguages={
      Object {
        "loadingLanguages": false,
        "message": "not loading",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
          Object {
            "fontSize": "1em",
            "height": "100%",
            "margin": "auto",
            "width": "100%",
          }
        }
      >
        <ResponsiveBar
          animate={true}
          axisBottom={null}
          axisLeft={null}
          axisRight={null}
          axisTop={null}
          borderColor="inherit:darker(1.6)"
          className="responsive-bar"
          colorBy="id"
          colors="set2"
          data={
            Array [
              Object {
                "CSS": "2.71",
                "CoffeeScript": "0.01",
                "HTML": "8.55",
                "JavaScript": "55.95",
                "Roff": "0.14",
                "Ruby": "32.63",
                "hover": "percent used",
              },
            ]
          }
          enableGridY={false}
          indexBy="hover"
          keys={
            Array [
              "Ruby",
              "hover",
              "CSS",
              "HTML",
              "Roff",
              "JavaScript",
              "CoffeeScript",
            ]
          }
          labelSkipWidth={12}
          labelTextColor="inherit:darker(1.6)"
          layout="horizontal"
          legends={
            Array [
              Object {
                "anchor": "bottom",
                "dataFrom": "keys",
                "direction": "row",
                "effects": Array [
                  Object {
                    "on": "hover",
                    "style": Object {
                      "itemOpacity": 1,
                    },
                  },
                ],
                "itemDirection": "left-to-right",
                "itemHeight": 20,
                "itemOpacity": 0.85,
                "itemWidth": 90,
                "itemsSpacing": 2,
                "justify": false,
                "symbolSize": 10,
                "translateX": 0,
                "translateY": 0,
              },
            ]
          }
          margin={
            Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
              "top": 0,
            }
          }
          motionDamping={15}
          motionStiffness={90}
          padding={0.3}
          reverse={false}
        />
      </div>,
      "className": "bar-holder",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ResponsiveBar
          animate={true}
          axisBottom={null}
          axisLeft={null}
          axisRight={null}
          axisTop={null}
          borderColor="inherit:darker(1.6)"
          className="responsive-bar"
          colorBy="id"
          colors="set2"
          data={
            Array [
              Object {
                "CSS": "2.71",
                "CoffeeScript": "0.01",
                "HTML": "8.55",
                "JavaScript": "55.95",
                "Roff": "0.14",
                "Ruby": "32.63",
                "hover": "percent used",
              },
            ]
          }
          enableGridY={false}
          indexBy="hover"
          keys={
            Array [
              "Ruby",
              "hover",
              "CSS",
              "HTML",
              "Roff",
              "JavaScript",
              "CoffeeScript",
            ]
          }
          labelSkipWidth={12}
          labelTextColor="inherit:darker(1.6)"
          layout="horizontal"
          legends={
            Array [
              Object {
                "anchor": "bottom",
                "dataFrom": "keys",
                "direction": "row",
                "effects": Array [
                  Object {
                    "on": "hover",
                    "style": Object {
                      "itemOpacity": 1,
                    },
                  },
                ],
                "itemDirection": "left-to-right",
                "itemHeight": 20,
                "itemOpacity": 0.85,
                "itemWidth": 90,
                "itemsSpacing": 2,
                "justify": false,
                "symbolSize": 10,
                "translateX": 0,
                "translateY": 0,
              },
            ]
          }
          margin={
            Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
              "top": 0,
            }
          }
          motionDamping={15}
          motionStiffness={90}
          padding={0.3}
          reverse={false}
        />,
        "style": Object {
          "fontSize": "1em",
          "height": "100%",
          "margin": "auto",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "animate": true,
          "axisBottom": null,
          "axisLeft": null,
          "axisRight": null,
          "axisTop": null,
          "borderColor": "inherit:darker(1.6)",
          "className": "responsive-bar",
          "colorBy": "id",
          "colors": "set2",
          "data": Array [
            Object {
              "CSS": "2.71",
              "CoffeeScript": "0.01",
              "HTML": "8.55",
              "JavaScript": "55.95",
              "Roff": "0.14",
              "Ruby": "32.63",
              "hover": "percent used",
            },
          ],
          "enableGridY": false,
          "indexBy": "hover",
          "keys": Array [
            "Ruby",
            "hover",
            "CSS",
            "HTML",
            "Roff",
            "JavaScript",
            "CoffeeScript",
          ],
          "labelSkipWidth": 12,
          "labelTextColor": "inherit:darker(1.6)",
          "layout": "horizontal",
          "legends": Array [
            Object {
              "anchor": "bottom",
              "dataFrom": "keys",
              "direction": "row",
              "effects": Array [
                Object {
                  "on": "hover",
                  "style": Object {
                    "itemOpacity": 1,
                  },
                },
              ],
              "itemDirection": "left-to-right",
              "itemHeight": 20,
              "itemOpacity": 0.85,
              "itemWidth": 90,
              "itemsSpacing": 2,
              "justify": false,
              "symbolSize": 10,
              "translateX": 0,
              "translateY": 0,
            },
          ],
          "margin": Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "top": 0,
          },
          "motionDamping": 15,
          "motionStiffness": 90,
          "padding": 0.3,
          "reverse": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "fontSize": "1em",
              "height": "100%",
              "margin": "auto",
              "width": "100%",
            }
          }
        >
          <ResponsiveBar
            animate={true}
            axisBottom={null}
            axisLeft={null}
            axisRight={null}
            axisTop={null}
            borderColor="inherit:darker(1.6)"
            className="responsive-bar"
            colorBy="id"
            colors="set2"
            data={
              Array [
                Object {
                  "CSS": "2.71",
                  "CoffeeScript": "0.01",
                  "HTML": "8.55",
                  "JavaScript": "55.95",
                  "Roff": "0.14",
                  "Ruby": "32.63",
                  "hover": "percent used",
                },
              ]
            }
            enableGridY={false}
            indexBy="hover"
            keys={
              Array [
                "Ruby",
                "hover",
                "CSS",
                "HTML",
                "Roff",
                "JavaScript",
                "CoffeeScript",
              ]
            }
            labelSkipWidth={12}
            labelTextColor="inherit:darker(1.6)"
            layout="horizontal"
            legends={
              Array [
                Object {
                  "anchor": "bottom",
                  "dataFrom": "keys",
                  "direction": "row",
                  "effects": Array [
                    Object {
                      "on": "hover",
                      "style": Object {
                        "itemOpacity": 1,
                      },
                    },
                  ],
                  "itemDirection": "left-to-right",
                  "itemHeight": 20,
                  "itemOpacity": 0.85,
                  "itemWidth": 90,
                  "itemsSpacing": 2,
                  "justify": false,
                  "symbolSize": 10,
                  "translateX": 0,
                  "translateY": 0,
                },
              ]
            }
            margin={
              Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
                "top": 0,
              }
            }
            motionDamping={15}
            motionStiffness={90}
            padding={0.3}
            reverse={false}
          />
        </div>,
        "className": "bar-holder",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ResponsiveBar
            animate={true}
            axisBottom={null}
            axisLeft={null}
            axisRight={null}
            axisTop={null}
            borderColor="inherit:darker(1.6)"
            className="responsive-bar"
            colorBy="id"
            colors="set2"
            data={
              Array [
                Object {
                  "CSS": "2.71",
                  "CoffeeScript": "0.01",
                  "HTML": "8.55",
                  "JavaScript": "55.95",
                  "Roff": "0.14",
                  "Ruby": "32.63",
                  "hover": "percent used",
                },
              ]
            }
            enableGridY={false}
            indexBy="hover"
            keys={
              Array [
                "Ruby",
                "hover",
                "CSS",
                "HTML",
                "Roff",
                "JavaScript",
                "CoffeeScript",
              ]
            }
            labelSkipWidth={12}
            labelTextColor="inherit:darker(1.6)"
            layout="horizontal"
            legends={
              Array [
                Object {
                  "anchor": "bottom",
                  "dataFrom": "keys",
                  "direction": "row",
                  "effects": Array [
                    Object {
                      "on": "hover",
                      "style": Object {
                        "itemOpacity": 1,
                      },
                    },
                  ],
                  "itemDirection": "left-to-right",
                  "itemHeight": 20,
                  "itemOpacity": 0.85,
                  "itemWidth": 90,
                  "itemsSpacing": 2,
                  "justify": false,
                  "symbolSize": 10,
                  "translateX": 0,
                  "translateY": 0,
                },
              ]
            }
            margin={
              Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
                "top": 0,
              }
            }
            motionDamping={15}
            motionStiffness={90}
            padding={0.3}
            reverse={false}
          />,
          "style": Object {
            "fontSize": "1em",
            "height": "100%",
            "margin": "auto",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "animate": true,
            "axisBottom": null,
            "axisLeft": null,
            "axisRight": null,
            "axisTop": null,
            "borderColor": "inherit:darker(1.6)",
            "className": "responsive-bar",
            "colorBy": "id",
            "colors": "set2",
            "data": Array [
              Object {
                "CSS": "2.71",
                "CoffeeScript": "0.01",
                "HTML": "8.55",
                "JavaScript": "55.95",
                "Roff": "0.14",
                "Ruby": "32.63",
                "hover": "percent used",
              },
            ],
            "enableGridY": false,
            "indexBy": "hover",
            "keys": Array [
              "Ruby",
              "hover",
              "CSS",
              "HTML",
              "Roff",
              "JavaScript",
              "CoffeeScript",
            ],
            "labelSkipWidth": 12,
            "labelTextColor": "inherit:darker(1.6)",
            "layout": "horizontal",
            "legends": Array [
              Object {
                "anchor": "bottom",
                "dataFrom": "keys",
                "direction": "row",
                "effects": Array [
                  Object {
                    "on": "hover",
                    "style": Object {
                      "itemOpacity": 1,
                    },
                  },
                ],
                "itemDirection": "left-to-right",
                "itemHeight": 20,
                "itemOpacity": 0.85,
                "itemWidth": 90,
                "itemsSpacing": 2,
                "justify": false,
                "symbolSize": 10,
                "translateX": 0,
                "translateY": 0,
              },
            ],
            "margin": Object {
              "bottom": 0,
              "left": 0,
              "right": 0,
              "top": 0,
            },
            "motionDamping": 15,
            "motionStiffness": 90,
            "padding": 0.3,
            "reverse": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`LanguageBarChart should match the snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LanguageBarChart
    languageData={Object {}}
    loadingLanguages={
      Object {
        "loadingLanguages": true,
        "message": "is loading",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "msg": "Looking up language usage",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": "Looking up language usage",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`LanguageBarChart should match the snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LanguageBarChart
    languageData={Object {}}
    loadingLanguages={
      Object {
        "loadingLanguages": true,
        "message": "is loading",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "msg": "Looking up language usage",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": "Looking up language usage",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
